src/entry.js

```diff
@@ -23,172 +29,226 @@ class Entry {
     if (!isDefined(data)) throw new Error('Entry requires data')
     if (!isDefined(next) || !Array.isArray(next)) throw new Error("'next' argument is not an array")
 
-    // Clean the next objects and convert to hashes
-    const toEntry = (e) => e.hash ? e.hash : e
+    // Clean the next objects and convert to cids
+    const toEntry = (e) => e.cid ? e.cid : e
     const nexts = next.filter(isDefined).map(toEntry)
 
     const entry = {
-      hash: null, // "Qm...Foo", we'll set the hash after persisting the entry
+      cid: null, // "zd...Foo", we'll set the cid after persisting the entry
       id: logId, // For determining a unique chain
       payload: data, // Can be any JSON.stringifyable data
-      next: nexts, // Array of Multihashes
-      v: 0, // For future data structure updates, should currently always be 0
+      next: nexts, // Array of CIDs
+      v: 1, // To tag the version of this data structure
       clock: clock || new Clock(identity.publicKey)
     }
 
     const signature = await identity.provider.sign(identity, Entry.toBuffer(entry))
+
     entry.key = identity.publicKey
     entry.identity = identity.toJSON()
     entry.sig = signature
-    entry.hash = await Entry.toMultihash(ipfs, entry)
+    entry.cid = await Entry.toCID(ipfs, entry)
     return entry
   }
 
   /**
-   * Verifies an entry signature for a given key and sig
-   * @param  {Entry}  entry Entry to verify
-   * @return {Promise}      Returns a promise that resolves to a boolean value
-   * indicating if the entry signature is valid
+   * Verifies an entry signature.
+   *
+   * @param {IdentityProvider} identityProvider The identity provider to use
+   * @param {Entry} entry The entry being verified
+   * @return {Promise} A promise that resolves to a boolean value indicating if the signature is valid
    */
   static async verify (identityProvider, entry) {
     if (!identityProvider) throw new Error('Identity-provider is required, cannot verify entry')
     if (!Entry.isEntry(entry)) throw new Error('Invalid Log entry')
     if (!entry.key) throw new Error("Entry doesn't have a key")
     if (!entry.sig) throw new Error("Entry doesn't have a signature")
 
-    const e = Object.assign({}, {
```

Comment:
Creating a new object was unnecessary, therefore I've removed the shallow clone.